src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt

```diff
@@ -133,15 +141,27 @@ class ElytraFlight : Module() {
 
                     if (!isStandingStill) packet.pitch = packetPitch
                 }
-                if (mode.value != ElytraFlightMode.CREATIVE) packet.yaw = packetYaw
+                if (effectiveMode != ElytraFlightMode.CREATIVE) packet.yaw = packetYaw
             }
         }
     })
 
     @EventHandler
     private val receiveListener = Listener(EventHook { event: PacketEvent.Receive ->
-        if (mc.player == null || mc.player.isSpectator || !elytraIsEquipped || elytraDurability <= 1 || !isFlying || mode.value == ElytraFlightMode.BOOST) return@EventHook
-        if (event.packet is SPacketPlayerPosLook && mode.value != ElytraFlightMode.PACKET) {
+        if (mc.player == null || mc.player.isSpectator || !elytraIsEquipped || elytraDurability <= 1 ) {
+            return@EventHook
+        }else{
+            if (pathfind&& (event.packet is SPacketEntityMetadata||
+                            event.packet is SPacketUpdateHealth||
+                            event.packet is SPacketEntityMetadata||
+                            event.packet is SPacketPlayerPosLook||
+                            event.packet is CPacketPlayer.Rotation||
+                            event.packet is CPacketPlayer.PositionRotation)) {
+                Astarpathfinder.onreceive(event)
+            }
+        }
+        if(!isFlying || effectiveMode == ElytraFlightMode.BOOST) return@EventHook
```

Comment:
I'll reformat this myself later, but it looks pretty bad :p

Reply:
you can leave out the packet checks but then every packet gets routed to the pathfinder. the ugly thing is that we need to handel the packets even when isFlying is false

Reply:
I meant the code style formatting itself